### PR TITLE
chore(deps)!: Update aws-cdk libraries to 1.145.0

### PR DESCRIPTION
**Pull request by akash1810**
BREAKING CHANGE: Update aws-cdk libraries to 1.145.0

### CODE REVIEW

**Review by jacobwinch — DISMISSED** [review]

File: src/constructs/rds/instance.test.ts [hunk 1]
```diff
@@ -75,7 +75,7 @@ describe("The GuDatabaseInstance class", () => {
 
     expect(stack).toHaveResource("AWS::RDS::DBInstance", {
       DBParameterGroupName: {
-        Ref: "DatabaseInstanceRDSParameterGroup307734CB",
+        Ref: "DatabaseInstanceTestingParameterGroup0F38B815",
```
[the remark this answers is not in the document]
Comment (akash1810) [reply src/constructs/rds/instance.test.ts:78]: Ah! Good spot! I guess we were broken before too, if a YAML template created a ParameterGroup with a different logical ID.

Feels like we want to accept a prop for the parameter group and override the automatic one when necessary?

[the remark this answers is not in the document]
Comment (akash1810) [reply src/constructs/rds/instance.test.ts:78]: @jacobwinch @nicl, a summary of our chat on this:
  - Great that AWS are doing this now
  - The change in logicalId to a `ParameterGroup` is not desirable as it likely causes service interruption
  - We should update GuCDK in a non-breaking, allowing teams to upgrade as a no-op, at the cost of maintaining some potentially fragile code

Before starting to work on this, I took a deeper look at our data (from https://github.com/guardian/cloudformation-information). There are 2 stacks using GuCDK that also create a `AWS::RDS::DBParameterGroup` resource:
  - Grafana (DevX owned)
  - Typerighter Rule Manager (Editorial Tools owned)

I'm pretty confident we can incur "some interruptions" on both of these stacks without any/much concern from their users.

Given this, I'd be inclined to _not_ update GuCDK in a no-op manner as the cost of maintaining the code would not be worth it. That is, keep/release the changes made in this PR.

Is that fair?

[the remark this answers is not in the document]
Comment (jacobwinch) [reply src/constructs/rds/instance.test.ts:78]: Thanks for summarising this. 

> I'd be inclined to not update GuCDK in a no-op manner as the cost of maintaining the code would not be worth it. That is, keep/release the changes made in this PR.

I'm happy to go with this approach - it's probably worth flagging this PR with the Ed Tools team as the change isn't obviously problematic just from looking at the snapshot diff (as an aside: I guess they could also override the logical id for the parameter group via an escape hatch if the service interruption was not acceptable to them?)

**Review by jacobwinch — APPROVED** [review]

File: src/constructs/rds/instance.test.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (akash1810) [reply src/constructs/rds/instance.test.ts:78]: [Testing](https://github.com/guardian/grafana/pull/121) this on Grafana didn't show any service interruption. The RDS metrics do not show any change, additionally created a script to refresh the home page of Grafana every second and saw no issue. However, it's worth noting that Grafana usage of RDS differs from a lot of other applications - IIUC Grafana only uses RDS for dashboard configuration, data is sourced from CW Metrics or ELK.

> it's probably worth flagging this PR with the Ed Tools team as the change isn't obviously problematic just from looking at the snapshot diff

👍🏽  Discussed in Chat with @jonathonherbert but tagging him here for explicitness.

> I guess they could also override the logical id for the parameter group via an escape hatch if the service interruption was not acceptable to them?

Ah, that's a good idea! Let's add this later if/when needed.

**Pull request merged**